commit 33fd8195529d74c0fe23cddd1c76fe7e03bbd324
Author: Johannes Berg <johannes@sipsolutions.net>
Date:   Mon Dec 1 18:50:27 2008 +0100

    iwlwifi: disable AP mode
    
    iwlwifi does not support AP mode in any way. For one, it doesn't even
    buffer multicast/broadcast frames properly. We didn't allow zd1211rw AP
    mode to be enabled without this, so iwlwifi shouldn't be allowed to
    advertise AP mode either.
    
    It also doesn't work at all, it doesn't even answer to probe requests,
    I'm guessing the packet injection code was disabled again.
    
    Signed-off-by: Johannes Berg <johannes@sipsolutions.net>
    Acked-by: Reinette Chatre <reinette.chatre@intel.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/drivers/net/wireless/iwlwifi/iwl3945-base.c b/drivers/net/wireless/iwlwifi/iwl3945-base.c
index 94ea4c236838..1d4658acba9c 100644
--- a/drivers/net/wireless/iwlwifi/iwl3945-base.c
+++ b/drivers/net/wireless/iwlwifi/iwl3945-base.c
@@ -7814,7 +7814,6 @@ static int iwl3945_pci_probe(struct pci_dev *pdev, const struct pci_device_id *e
 		    IEEE80211_HW_NOISE_DBM;
 
 	hw->wiphy->interface_modes =
-		BIT(NL80211_IFTYPE_AP) |
 		BIT(NL80211_IFTYPE_STATION) |
 		BIT(NL80211_IFTYPE_ADHOC);